[TensorExpr Fuser] Add support for nodes which have tensor constant inputs

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48264 [NNC] Preserve strided output
* #48263 Remove inferred from tensor type ctors
* **#47814 [TensorExpr Fuser] Add support for nodes which have tensor constant inputs**
* #47813 [NNC] Compute Tensor Output Properties in ininitialization
* #47812 [TensorExpr] Cache use of fallback in kernel invocation
* #47810 Dont use symbolic shapes check

Previously, we would bail completely if a node had a constant tensor input. This PR adds support for this case by lifting the constant out of the fusion graph after we've done fusion. It might be nice to add support for Tensor Constants in NNC itself, but it looked kind of tricky and this is an easy enough temporary solution.

Differential Revision: [D25286215](https://our.internmc.facebook.com/intern/diff/D25286215)

Commit:
Add support for nodes which ahve tensor constant inputs

[ghstack-poisoned]

Commit:
Update on "Add support for nodes which ahve tensor constant inputs"




[ghstack-poisoned]

ZolotukhinM commented on test/jit/test_profiler.py:
```diff
@@ -193,6 +195,19 @@ def foo(a, b):
+
+        self.assertEqual(foo_script(x, x), foo(x, x))
+        g = torch.jit.last_executed_optimized_graph()
+        FileCheck().check_count("aten::add", 2, exactly=True).run(g)
```
Do we want to check that these adds are in the fusion group? Otherwise this test will pass even if we fuse nothing.

eellison replied:
lol good point

ZolotukhinM reviewed (approved):
Nice!

Commit:
Update on "[TensorExpr Fuser] Add support for nodes which have tensor constant inputs"


Previously, we would bail completely if a node had a constant tensor input. This PR adds support for this case by lifting the constant out of the fusion graph after we've done fusion. It might be nice to add support for Tensor Constants in NNC itself, but it looked kind of tricky and this is an easy enough temporary solution. 

[ghstack-poisoned]

Commit:
Update on "[TensorExpr Fuser] Add support for nodes which have tensor constant inputs"


Previously, we would bail completely if a node had a constant tensor input. This PR adds support for this case by lifting the constant out of the fusion graph after we've done fusion. It might be nice to add support for Tensor Constants in NNC itself, but it looked kind of tricky and this is an easy enough temporary solution. 

[ghstack-poisoned]

Commit:
Update on "[TensorExpr Fuser] Add support for nodes which have tensor constant inputs"


Previously, we would bail completely if a node had a constant tensor input. This PR adds support for this case by lifting the constant out of the fusion graph after we've done fusion. It might be nice to add support for Tensor Constants in NNC itself, but it looked kind of tricky and this is an easy enough temporary solution.

Differential Revision: [D25286215](https://our.internmc.facebook.com/intern/diff/D25286215)

[ghstack-poisoned]
